sympy/functions/elementary/trigonometric.py

```diff
@@ -305,6 +305,8 @@ def eval(cls, arg):
 
             if (2*pi_coeff).is_integer:
                 if pi_coeff.is_even:
+                    # Probably never happens since pi_coeff.is_integer is
+                    # tested above
```

Comment:
These is_even cases can just be removed

Reply:
(although a comment could be added for clarification)

Reply:
Good that you agree. Didn't want to remove them in case i missed something. Will update this (and probably add a few more tests...).

Reply:
It seems that these lines were added to fix #8298 in case `arg` is `pi*n/2` with `n.is_integer`. But it seems odd that `pi_coeff` (which is `n/2` in this case) should be even.

Reply:
Thanks for looking it up!

Probably the assumption system has improved since.

```
x = Symbol('x', even=True)
(x/2).is_integer
```
now returns `True` which it probably didn't then.

Reply:
I suspect that there was some confusion between `n` and `pi_coeff`. It looks like `if pi_coeff.is_even:` was written with `n` in mind.